### PR TITLE
Fix missing features and bugs for nutanix csi

### PR DESCRIPTION
**Pull request by mate4st**
**What does this PR do / Why do we need it**:

- Fixes missing dereference for nutanix port pointer
- Fixes missing default storage class
- Allow storageclass configuration

```release-note
NONE
```

### CODE REVIEW

File: addons/default-storage-class/storage-class.yaml [hunk 1]
```diff
@@ -124,3 +124,26 @@ provisioner: csi.hetzner.cloud
...
+provisioner: csi.nutanix.com
+parameters:
+  storageType: NutanixVolumes
+  csi.storage.k8s.io/provisioner-secret-name: ntnx-secret
+  csi.storage.k8s.io/provisioner-secret-namespace: kube-system
+  csi.storage.k8s.io/node-publish-secret-name: ntnx-secret
+  csi.storage.k8s.io/node-publish-secret-namespace: kube-system
+  csi.storage.k8s.io/controller-expand-secret-name: ntnx-secret
+  csi.storage.k8s.io/controller-expand-secret-namespace: kube-system
+  storageContainer: {{ default "Default" .Params.storageContainer | quote }}
```
Comment (embik) [inline addons/default-storage-class/storage-class.yaml]: Can a user even change this?

Comment (mate4st) [reply addons/default-storage-class/storage-class.yaml]: No this is a copy paste leftover from kubeone. I thought i removed all

File: pkg/addon/template.go [hunk 2]
```diff
@@ -216,6 +218,18 @@ type MLASettings struct {
...
+
+	//vsphere
+	// StoragePolicy is the storage policy to use for vsphere csi addon
+	StoragePolicy string
+
+	//nutanix
+	StorageContainer        string
+	Fstype                  string
+	SsSegmentedIscsiNetwork *bool
+}
```
Comment (embik) [inline pkg/addon/template.go]: Can we maybe change this to something like:

```go
type CSIOptions struct {
     Vsphere *VsphereCSIOptions
     Nutanix *NutanixCSIOptions
}

type VsphereCSIOptions struct {
    StoragePolicy string
}

type NutanixCSIOptions struct {
    StorageContainer        string
    Fstype                  string
    SsSegmentedIscsiNetwork *bool
}
```

I don't think it's a good idea to have a shared struct holding all provider-specific options.

**Review by embik — CHANGES_REQUESTED** [review]

File: pkg/addon/template.go (hunk 2 above)
Comment (mate4st) [reply pkg/addon/template.go]: As this is only used for the templating and for now we do not have a lot of these features its okay in my opinion.

Comment (embik) [reply pkg/addon/template.go]: Not a huge fan of it to be honest, but okay :D

**Review by embik — APPROVED** [review]
/approve
/lgtm

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**